app/controllers/MarriageAllowanceController.scala

```diff
@@ -40,14 +40,17 @@ trait MarriageAllowanceController extends BaseController {
 
   val authAction: AuthAction
 
+  //TODO refactor so that the correct response are being sent
   def getRecipientRelationship(transferorNino: Nino) = authAction.async(parse.json) { implicit request =>
     withJsonBody[FindRecipientRequest] { findRecipientRequest =>
       marriageAllowanceService.getRecipientRelationship(transferorNino, findRecipientRequest) map {
-        case (recipientRecord: UserRecord, taxYears: List[TaxYear]) =>
+        case Right((recipientRecord, taxYears)) =>
           Ok(Json.toJson(GetRelationshipResponse(
             user_record = Some(recipientRecord),
             availableYears = Some(taxYears),
             status = ResponseStatus(status_code = "OK"))))
+        case Left(error: DataRetrievalError) => Ok(Json.toJson(GetRelationshipResponse(
```

Comment:
We are not using the DataRetrievalError here. Is it correct to send a RECIPIENT_NOT_FOUND in every error case?

Reply:
This needs to be cleaned up here and in TAMC_FRONTEND under a separate ticket.